src/main/webapp/creatives/maps/simid_map_creative.js

```diff
@@ -20,17 +21,55 @@ export default class SimidMapCreative extends BaseSimidCreative {
   }
 
   /** @override */
-  onStart(eventData){
-    super.onStart(eventData);
-    //ToDo(juliareichel@): handle invalid JSON and param errors
-    const adParams = JSON.parse(this.creativeData.adParameters);
+  onInit(eventData) {
+    this.creativeData = eventData.args.creativeData;
+    this.environmentData = eventData.args.environmentData;
+
+
+    this.videoState = {
+      currentSrc:'',
+      currentTime: -1, // Time not yet known
+      duration: -1, // duration unknown
+      ended: false,
+      muted: this.environmentData.muted,
+      paused: false,
+      volume: this.environmentData.volume,
+      fullscreen: false,
+    }
+
+    /**
+     * Error handeling for empty or incorrect ad paramters.
+     */
+    let adParams = "";
```

Comment:
This function has gotten a little complex.
How about breaking out all validation code into a separate function?

Reply:
done